src/invidious/yt_backend/extractors.cr

```diff
@@ -718,8 +718,11 @@ private module Extractors
       raw_items = [] of Array(JSON::Any)
 
       target.dig("primaryContents", "sectionListRenderer", "contents").as_a.each do |node|
-        if node = node["itemSectionRenderer"]?
-          raw_items << node["contents"].as_a
+        if new_node = node["itemSectionRenderer"]?
+          raw_items << new_node["contents"].as_a
+        end
+        if node["continuationItemRenderer"]?
+          raw_items.push([node])
         end
```

Comment:
```suggestion
        if new_node = node["itemSectionRenderer"]?
          raw_items << new_node["contents"].as_a
        elsif node["continuationItemRenderer"]?
          raw_items.push([node])
        end
```

Reply:
Iirc they're both in the same object which is why i didnt do an elsif. I can probably double check in a few days (not at my computer right now)

Reply:
I'm wondering, what's the use case for that?
If I understand correctly, it will extract the continuation for shelf renderers and alike?

Reply:
Yeah, for the search page, the continuation seems to be in the same json object as the itemSectionRenderer

Reply:
@syeopite improving upon your suggestion with the use of `<<`
```suggestion
        if new_node = node["itemSectionRenderer"]?
          raw_items << new_node["contents"].as_a
        elsif node["continuationItemRenderer"]?
          raw_items << node
        end
```